test-runner-poc/lib/runner.js

```diff
@@ -29,18 +28,28 @@ function environmentSetup(environmentCallback, done) {
     createFilters(environmentParser);
     function environmentParser(environmentName, version) {
       const Environment = environments[environmentName];
-      client.db('admin').command({ isMaster: 1 }, (err, result) => {
+
+      function callEnvironmentCallback(result, version, err) {
+        const environment = new Environment(result, version);
+        environment.mongo = require('../../index');
+        environmentCallback(err, environment);
+        return client.close(done);
```

Comment:
Why do we need `done` here?

Reply:
`environmentCallback` does not call `done` unless there is an error

Reply:
But this isn't running inside of a test, correct?

Reply:
the `before` function has `done` passed in as a parameter. Then, it calls `environmentSetup` with the same `done` passed in as a parameter. This was to ensure that only after the `environmentSetup` is finished running, will `before` call `done`.

Reply:
We used `return client.close(done)` in order to allow the `before` hook to call done and finish.

Reply:
oh gotcha, thanks!